python-package/lightgbm/sklearn.py

```diff
@@ -656,6 +656,7 @@ def fit(self, X, y,
             eval_class_weight=None, eval_init_score=None, eval_metric=None,
             early_stopping_rounds=None, verbose=True,
             feature_name='auto', categorical_feature='auto', callbacks=None):
+        _LGBMAssertAllFinite(y)
```

Comment:
Fix the following error in `estimator_checks`:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
C:\Program Files\Anaconda3\lib\site-packages\sklearn\utils\estimator_checks.py in check_supervised_y_no_nan(name, estimator_orig)
    162     try:
--> 163         estimator.fit(X, y)
    164     except ValueError as e:

C:\Program Files\Anaconda3\lib\site-packages\lightgbm\sklearn.py in fit(self, X, y, sample_weight, init_score, eval_set, eval_names, eval_sample_weight, eval_class_weight, eval_init_score, eval_metric, early_stopping_rounds, verbose, feature_name, categorical_feature, callbacks)
    685             feature_name='auto', categorical_feature='auto', callbacks=None):
--> 686         _LGBMCheckClassificationTargets(y)
    687         self._le = _LGBMLabelEncoder().fit(y)

C:\Program Files\Anaconda3\lib\site-packages\sklearn\utils\multiclass.py in check_classification_targets(y)
    170                       'multilabel-indicator', 'multilabel-sequences']:
--> 171         raise ValueError("Unknown label type: %r" % y_type)
    172 

ValueError: Unknown label type: 'continuous'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-2-accbb017d982> in <module>()
      9             continue  # skip test because LightGBM deals with nan
     10         try:
---> 11             check(name, estimator)
     12         except SkipTest as message:
     13             warnings.warn(message, SkipTestWarning)

C:\Program Files\Anaconda3\lib\site-packages\sklearn\utils\testing.py in wrapper(*args, **kwargs)
    348             with warnings.catch_warnings():
    349                 warnings.simplefilter("ignore", self.category)
--> 350                 return fn(*args, **kwargs)
    351 
    352         return wrapper

C:\Program Files\Anaconda3\lib\site-packages\sklearn\utils\estimator_checks.py in check_supervised_y_no_nan(name, estimator_orig)
    166             raise ValueError("Estimator {0} raised error as expected, but "
    167                              "does not match expected error message"
--> 168                              .format(name))
    169     else:
    170         raise ValueError("Estimator {0} should have raised error on fitting "

ValueError: Estimator LGBMClassifier raised error as expected, but does not match expected error message
```